src/test/ensemble_tests.cpp

```diff
@@ -2195,3 +2195,146 @@ TEST_F(EnsembleFlowTest, WaitForLoadingPipelineDefinitionFromBeginStatus) {
     t.join();
     t2.join();
 }
+
+const char* configJsonWithNoPipeline = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"all": {}},
+                "nireq": 1
+            }
+        }
+    ]
+})";
+
+TEST_F(EnsembleFlowTest, RetirePipelineAfterLoading) {
+    std::string fileToReload = "/tmp/ovms_config_file1.json";
+    createConfigFileWithContent(pipelineOneDummyConfig, fileToReload);
+    ConstructorEnabledModelManager manager;
+    auto status = manager.startFromFile(fileToReload);
+    manager.startWatcher();
+    ASSERT_TRUE(status.ok()) << status.string();
+    ASSERT_EQ(manager.getPipelineFactory().findDefinitionByName(PIPELINE_1_DUMMY_NAME)->getStatusCode(),
+        PipelineDefinitionStateCode::AVAILABLE);
+    createConfigFileWithContent(configJsonWithNoPipeline, fileToReload);
+    std::this_thread::sleep_for(std::chrono::milliseconds(1500));
```

Comment:
Is this the end of that test? Shouldn't we check if after reload we got expected state?

Reply:
I missed that ... Thanks!